treeshr/treeshrp.h

```diff
@@ -397,14 +397,54 @@ typedef struct
 } RFA;
 
 #ifdef RFA_MACROS
-#define RfaToSeek(rfa)                  \
-  (((*(unsigned int *)rfa - 1) * 512) + \
-   (*(unsigned short *)&((char *)rfa)[4] & 0x1ff))
-#define SeekToRfa(seek, rfa)                                               \
-  {                                                                        \
-    *(unsigned int *)rfa = (unsigned int)(seek / 512 + 1);                 \
-    *(unsigned short *)&(((char *)rfa)[4]) = (unsigned short)(seek % 512); \
-  }
+
+#include <stdint.h>
+/*
+ * RFA layout (byte-packed, little-endian):
+ * rfa[0..3] : uint32_t block_number_1based
+ * rfa[4..5] : uint16_t offset (low 9 bits used)
+ *
+ * seek = (block - 1) * 512 + (offset & 0x1FF)
+ */
+
+static inline uint32_t rfa_to_seek(const RFA *r)
```

Comment:
I'm not sure what happened here, but we need to be incredibly careful with this code. This seems to have renamed the macros (which are only defined if `RFA_MACROS` is defined, which it never is) and replaced them with functions. We could verify these functions output the same thing as their macro counterparts but I think better yet we should just remove the entire section relating to `RFA_MACROS` and trust the functions defined in `TreeGetNci.c`